File: 0001. Two Sum/solution_2.cpp
class Solution 
{
    public:
        vector<int> twoSum(vector<int>& nums, int target) 
        {
            map<int, int> numsAndIndex;

            for (int i = 0; i < size(nums); ++i)
            {
                int curr = nums[i];
                int diff = target - nums[i];

                if (numsAndIndex.find(diff) != numsAndIndex.end())
                    return {numsAndIndex[diff], i};
                
                numsAndIndex[curr] = i;
            }

            return {};
        }
};

File: leetcode/0-9/1. Two Sum/solution_2.java
class Solution 
{
    public int[] twoSum(int[] nums, int target) 
    {
        HashMap<Integer, Integer> numAndIndex = new HashMap<Integer, Integer>();

        for (int i = 0; i < nums.length; ++i)
        {
            int curr = nums[i];
            int diff = target - curr;

            if (numAndIndex.containsKey(diff))
                return new int[]{numAndIndex.get(diff), i};

            numAndIndex.put(curr, i);
        }

        return new int[]{};
    }
}